modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java

```diff
@@ -61,12 +67,18 @@ public void addClusterSettingsListeners(
     }
 
     /**
-     * Copies APM settings from the provided settings object into the corresponding system properties.
+     * Initialize APM settings from the provided settings object into the corresponding system properties.
+     * Later updates to these settings are synchronized using update consumers.
      * @param settings the settings to apply
      */
-    public void syncAgentSystemProperties(Settings settings) {
-        this.setAgentSetting("recording", Boolean.toString(APM_ENABLED_SETTING.get(settings)));
```

Comment:
This looks also problematic, `recording` was previously only enabled if tracing was when initially synching the agent settings.

That makes me wonder if this is needed at all? It looks like the update consumers are triggering at startup which would then typically enable `recording` (assuming metrics are generally enabled).